srf_generation/input_file_generation/realisation_to_srf.py

```diff
@@ -392,6 +393,161 @@ def get_n(fault_size, sub_fault_size):
+
+    if (
+        utils.compare_versions(genslip_version, "5.4.2") < 0
+        and tect_type == "SUBDUCTION_INTERFACE"
+    ):
+        raise RuntimeError(
+            "Subduction interface faults are only available for version 5.4.2 and above"
+        )
+
+    gsf_file = NamedTemporaryFile(mode="w", delete=False)
+    logger.debug(f"Gsf will be saved to the temporary file {gsf_file.name}")
+    srf_file = realisation_file.replace(".csv", ".srf")
+    logger.debug(f"Srf will be saved to {srf_file}")
+    corners_file = realisation_file.replace(".csv", ".corners")
+    logger.debug(f"The corners file will be saved to {corners_file}")
+
+    with NamedTemporaryFile(mode="w", delete=False) as gsfp:
+        logger.debug("Saving segments file to {}".format(gsfp.name))
+        gsfp.write("{}\n".format(plane_count))
+        for f in range(plane_count):
+            gsfp.write(
+                "{:f} {:f} {:f} {:.4f} {:.4f} {:.4f} {:.4f} {:.4f} {:d} {:d}\n".format(
+                    clon[f],
+                    clat[f],
+                    dtop,
+                    strike[f],
+                    dip,
+                    rake,
+                    flen[f],
+                    fwid[f],
+                    nx[f],
+                    ny,
+                )
+            )
+
+    fault_seg_bin = binary_version.get_unversioned_bin(FAULTSEG2GSFDIPDIR)
+    cmd = [
+        fault_seg_bin,
+        "read_slip_vals=0",
+        "infile={}".format(gsfp.name),
+        "outfile={}".format(gsf_file.name),
+    ]
+    if dip_dir is not None:
+        cmd.append("dipdir={}".format(dip_dir))
+
+    logger.info("Calling fault_seg2gsf_dipdir with command {}".format(" ".join(cmd)))
+    gexec = run(cmd)
+    logger.debug(f"{fault_seg_bin} finished running with stderr: {gexec.stderr}")
+
+    remove(gsfp.name)
+    logger.debug("Removed segments file")
+
+    if int(plane_count > 1):
+        logger.debug("Multiple segments detected. Generating xseg argument")
+        flen_array = np.asarray(flen)
+        xseg = ",".join(map(str, flen_array.cumsum() - flen_array / 2))
+    else:
+        xseg = "-1"
+
+    gen_srf(
+        srf_file,
+        gsf_file.name,
+        4,
+        magnitude,
+        dt,
+        sum(nx),
+        ny,
+        seed,
+        shypo,
+        dhypo,
+        vel_mod_1d,
+        genslip_version=genslip_version,
+        rvfac=rvfac,
+        xseg=xseg,
+        logger=logger,
+    )
+
+    logger.info("srf generated, creating stoch")
+
+    if stoch_file is None:
+        stoch_file = realisation_file.replace(".csv", ".stoch")
+    create_stoch(stoch_file, srf_file, single_segment=(plane_count == 1), logger=logger)
+
+    logger.info("stoch created, making info")
+
+    # save INFO
+    create_info_file(
+        srf_file,
+        4,
+        magnitude,
+        rake,
+        dt,
+        tect_type=tect_type,
+        dip_dir=dip_dir,
+        shypo=[shypo + 0.5 * flen[i] for i in range(plane_count)],
```

Comment:
I'm not 100% sure that's the behaviour we want? I guess doing a test simulation and looking at the quick animation should make sure we have the fault planes propogating correctly.

Reply:
I believe that's the prior behaviour? Will double check

Reply:
why are we assuming something here though? since the models are already generated, can we not read it?

Reply:
https://github.com/ucgmsim/Pre-processing/blob/04de03a004d4dd0e487369aa44c392c610b965b4/SrfGen/NHM/nhm2srf.py#L406 // The original behaviour

Reply:
The only thing that seems to read shyp0 from the info file is plot_nhm_outputs, so the CS sources should be ok

Reply:
Original here: https://github.com/ucgmsim/Pre-processing/blob/b2aa5267e243dd68b15e551733d1075b80b79220/SrfGen/NHM/nhm2srf.py#L406-L409